ion_derive/src/template.rs

```diff
@@ -17,61 +17,219 @@ pub fn emit_template(ast: &DeriveInput) -> TokenStream {
+
+            impl #name {
+                pub fn from_name(config: &Config, name: &String) -> Result<Self> {
+                    let mut template = config.template_dir();
+                    template.push(name);
+                    template.push("template.toml");
+
+                    assert!(template.is_file(), "Template file not found: {}", template.display());
+                    let source = std::fs::read_to_string(template)?;
+                    let template : Template = toml::from_str(&source)?;
+                    Ok(template)
+                }
+
+                pub fn render(&self, config: &Config, ctx: &mut Context) -> Result<()> {
+                    let old_pwd = std::env::current_dir()?;
+                    std::env::set_current_dir(&*ctx.project.path)?;
+
+                    self.collect(config, ctx)?;
+                    debug!("Context: {:#?}", ctx);
+                    if ctx.prompt {
+                        self.prompt(config, ctx)?;
+                    }
+                    #render
+                    self.post_render(config, ctx)?;
+                    self.validate(config, ctx)?;
+
+                    std::env::set_current_dir(old_pwd)?;
+                    Ok(())
+                }
+
+                pub fn collect(&self, config: &Config, ctx: &mut Context) -> Result<()> {
+                    #collect
+                    Ok(())
+                }
 
-            pub fn render(&self, config: &Config, ctx: &mut Context) -> Result<()> {
-                let old_pwd = std::env::current_dir()?;
-                std::env::set_current_dir(&*ctx.project.path)?;
+                pub fn prompt(&self, config: &Config, ctx: &mut Context) -> Result<()> {
+                    #prompt
+                    Ok(())
+                }
+
+                pub fn post_render(&self, config: &Config, ctx: &Context) -> Result<()> {
+                    #post_render
+                    Ok(())
+                }
 
-                self.collect(config, ctx)?;
-                debug!("Context: {:#?}", ctx);
-                if ctx.prompt {
-                    self.prompt(config, ctx)?;
+                pub fn validate(&self, config: &Config, ctx: &Context) -> Result<()> {
+                    #validate
+                    Ok(())
                 }
-                #render
-                self.post_render(config, ctx)?;
-                self.validate(config, ctx)?;
 
-                std::env::set_current_dir(old_pwd)?;
-                Ok(())
             }
 
-            pub fn collect(&self, config: &Config, ctx: &mut Context) -> Result<()> {
-                #collect
-                Ok(())
+
+    impl fmt::Display for Template {
+        fn fmt(&self, format_buffer: &mut fmt::Formatter) -> fmt::Result {
+            write!(
+                format_buffer,
+                "Name:\n{}\n\nDescription:\n{}\n\n",
+                self.name, self.description
+            )?;
+
+            if let Some(repo) = &self.repo {
```

Comment:
you need to generate the names instead of manually writing them, like how I generate other methods

Reply:
Yeah, perhaps we should just do the debug printing, so that you don't need to implement this now